src/Routing/Router.php

```diff
@@ -655,7 +655,13 @@ public static function reverseToArray($params): array
             $params['_matchedRoute'],
             $params['_name']
         );
-        $params = array_merge($params, $pass);
+        foreach ($pass as $i => $passedValue) {
+            // Remove passed values that are also route keys
+            if (in_array($passedValue, $params)) {
```

Comment:
Do we need
```suggestion
            if (in_array($passedValue, $params, true)) {
```
here to avoid side effects of matching with numeric values and strings or alike?